### PR TITLE
Add dependencyStream using GraphQL

### PR DESCRIPTION
**Pull request by ericboucher**
Add dependencyStream using graphql

I am mostly focusing on extracting repo information (see schema) but let me know if you think we should fetch more data.

### CODE REVIEW

File: tap_github/repository_streams.py [hunk 1]
```diff
@@ -1953,3 +1953,114 @@ def _write_schema_message(self):
         if not self._schema_emitted:
             super()._write_schema_message()
             self._schema_emitted = True
+
+
+class DependencyStream(GitHubGraphqlStream):
```
Comment (laurentS) [inline tap_github/repository_streams.py:1958]: Maybe use the plural? `DependenciesStream` just for consistency with *most* other streams (we've not actually been very consistent...)

File: tap_github/repository_streams.py [hunk 2]
```diff
@@ -1953,3 +1953,114 @@ def _write_schema_message(self):
...
+    """Defines 'UserContributedToStream' stream. Warning: this stream 'only' gets the first 100 projects (by stars)."""
```
Comment (laurentS) [inline tap_github/repository_streams.py:1959]: leftovers from copy/paste?

File: tap_github/repository_streams.py [hunk 3]
```diff
@@ -1953,3 +1953,114 @@ def _write_schema_message(self):
...
+        """
+        headers = super().http_headers
+        headers["Accept"] = "application/vnd.github.hawkgirl-preview+json"
+        return headers
+
+    def post_process(self, row: dict, context: Optional[Dict] = None) -> dict:
+        """
+        Add a user_id top-level field to be used as state replication key.
+        """
+        print(row)
```
Comment (laurentS) [inline tap_github/repository_streams.py:1984]: debugging?

File: tap_github/repository_streams.py [hunk 4]
```diff
@@ -1953,3 +1953,114 @@ def _write_schema_message(self):
...
+
+        Overridden to use the preview for Repository.dependencyGraphManifests.
+        """
+        headers = super().http_headers
+        headers["Accept"] = "application/vnd.github.hawkgirl-preview+json"
+        return headers
+
+    def post_process(self, row: dict, context: Optional[Dict] = None) -> dict:
+        """
+        Add a user_id top-level field to be used as state replication key.
```
Comment (laurentS) [inline tap_github/repository_streams.py:1982]: leftovers from copy/paste?
```suggestion
        Add a dependency_repo_id top-level field to be used as primary key.
```

**Review by laurentS — APPROVED** [review]

File: tap_github/repository_streams.py (hunk 3 above)
Comment (ericboucher) [reply tap_github/repository_streams.py:1984]: Can't see that, are you looking at the latest?

**Review by edgarrmondragon — APPROVED** [review]
lgtm!

**Pull request merged**